packages/eslint-plugin/docs/rules/naming-convention.md

```diff
@@ -203,6 +203,9 @@ Individual Selectors match specific, well-defined sets. There is no overlap betw
 - `accessor` - matches any accessor.
   - Allowed `modifiers`: `abstract`, `override`, `private`, `protected`, `public`, `requiresQuotes`, `static`.
   - Allowed `types`: `array`, `boolean`, `function`, `number`, `string`.
+- `autoAccessor` - matches any auto-accessor.
+  - Allowed `modifiers`: `abstract`, `override`, `private`, `protected`, `public`, `requiresQuotes`, `static`.
+  - Allowed `types`: `array`, `boolean`, `function`, `number`, `string`.
```

Comment:
[Docs] I'm getting confused reviewing this and I'm a maintainer & have read the TC39 specs for much of these... so I can only imagine how mind-boggingly confusing the difference between _"accessor"_ (as the rule already calls things) and _"autoAccessor"_ (what's being added) is. Which is not your "fault" as the PR author, just ... a fact of how convoluted this rule has gotten.

I think this docs page will need some note explaining the nomenclature. People are going to get tripped up on the differences between the two, and get mad at us for having such confusing names. What do you think about explaining them briefly somewhere around this list?

At the very least I think we'd need definitions somewhere of what "accessor" and "autoAccessor" each mean. Many developers won't about the [`accessor` syntax](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#auto-accessors-in-classes).

Reply:
I think the distinction is "oaky" if you know the syntax:
- "accessors" are the classic top-level get/set accessors
- "auto accessors" are the members declared with the `accessor` keyword

```ts
class Foo {
  // accessor
  get a() { return 1 }
  set a(val: number) { }

  // auto-accessor field from the decorators proposal (supported in TS)
  accessor b: number = 1;

  // stage 1 auto-accessors proposal (not supported in TS)
  accessor c { get; set }
  accessor d { get; set } = 1;
  accessor e { get() { return 1 } }
  accessor f { set(val) { } }
  accessor g { #get; set }
  // etc
}
```

There's probably scope here to re-define `accessor` as a group selector that encompasses `autoAccessor`.

EG like:
`classicAccessor` = just the top-level, non-keyworded accessor syntax
`autoAccessor` = the new keyworded syntax
`accessor` = group of `classicAccessor | autoAccessor`

WDYT?

Reply:
> re-define `accessor` as a group selector

Ye, nice point.

Reply:
Brad's suggestion sounds good to me!